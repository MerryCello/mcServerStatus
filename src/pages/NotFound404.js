import React from "react";
import { useRouteError } from "react-router-dom";

const NotFound404 = () => {
  const error = useRouteError();
  console.log(error);
  return (
    <div className="main-container delete-page-container column">
      <p>404: Not Found</p>
      <p className="m-0">~~~~~~~~~~~~~~~~~~~</p>
      <p>The page you were looking for is in another realm</p>
    </div>
  );
};

export default NotFound404;
