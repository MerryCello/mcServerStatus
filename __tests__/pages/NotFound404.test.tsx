/** @jest-environment jsdom */
import React from 'react';
import {render} from '@testing-library/react';
import NotFound404 from '../../src/pages/NotFound404';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useRouteError: jest.fn(),
}));

describe('NotFound404', () => {
  test('renders 404 message', () => {
    const {container, getByText} = render(<NotFound404 />);
    expect(getByText('404: Not Found')).toBeInTheDocument();
    expect(
      getByText('The page you were looking for is in another realm'),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  test('renders decorative line', () => {
    const {getByText} = render(<NotFound404 />);
    expect(getByText('~~~~~~~~~~~~~~~~~~~')).toBeInTheDocument();
  });
});
