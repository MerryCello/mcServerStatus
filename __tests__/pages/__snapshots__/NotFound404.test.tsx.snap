// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound404 renders 404 message 1`] = `
<div>
  <div
    class="main-container delete-page-container column"
  >
    <p>
      404: Not Found
    </p>
    <p
      class="m-0"
    >
      ~~~~~~~~~~~~~~~~~~~
    </p>
    <p>
      The page you were looking for is in another realm
    </p>
  </div>
</div>
`;
